src/util.ts

```diff
@@ -421,3 +421,56 @@ interface IUploadResult {
     mxcUrl: string;
     size: number;
 }
+
+// Type type
+type Type = Function;  // tslint:disable-line ban-types
+
+/**
+ * Returns true if `obj` is subtype of at least one of the given types.
+ */
+export function instanceofsome(obj: object, types: Type[]): boolean {
+    return types.some((type) => obj instanceof type);
+}
+
+/**
+ * Append the old error message to the new one and keep its stack trace.
+ * Example:
+ *     throw wrap(e, HighLevelError, "This error is more specific");
+ */
+export function wrap<T extends Error>(
+    oldError: Type,
+    newErrorType: new (...args: any[]) => T,  // tslint:disable-line no-any
+    ...args: any[]  // tslint:disable-line no-any trailing-comma
+): T {
+    const newError = new newErrorType(...args);
+    let appendMsg;
+    if (oldError instanceof Error) {
+        appendMsg = oldError.message;
+        newError.stack = oldError.stack;
+    } else {
+        appendMsg = oldError.toString();
+    }
+    newError.message += ":\n" + appendMsg;
+    return newError;
+}
+
+/**
+ * Returns the guild and channel of the given remote room extracted from its ID.
+ * @param remoteRoom The room from which to get the guild and channel.
+ * @returns (guild, channel)-tuple.
+ */
+export function guildAndChannelOf(remoteRoom: RemoteRoom): [string, string] {
+    const ROOM_NAME_PARTS = 2;
+    return remoteRoom.roomId.substr("_discord".length).split("_", ROOM_NAME_PARTS);
+}
+
+/**
+ * Returns true if the given event is a bot command.
+ */
+export function isBotCommand(event: IMatrixEvent): boolean {
```

Comment:
We only really use this in one class, so I'd make it a private function of that class.

Reply:
It's not so much class related, thus I made it a module-scoped function. (It would be best to have it as method for the (hypothetical) MatrixMsgEvent class, but it's not a class and JS/TS don't have nice ways to do that extension anyways :disappointed: )